File: app/src/main/java/com/kaciula/archiman/injection/DomainModule.java
package com.kaciula.archiman.injection;

import com.kaciula.archiman.domain.boundary.data.AppInfoRepository;
import com.kaciula.archiman.domain.boundary.data.UserRepository;
import com.kaciula.archiman.domain.usecases.GetUsers;
import com.kaciula.archiman.domain.usecases.InitColdStart;
import com.kaciula.archiman.domain.util.SchedulerProvider;
import dagger.Module;
import dagger.Provides;
import javax.inject.Singleton;

@Module
public final class DomainModule {

  @Provides
  @Singleton
  GetUsers provideGetUsers(UserRepository userRepository, SchedulerProvider schedulerProvider) {
    return new GetUsers(userRepository, schedulerProvider);
  }

  @Provides
  @Singleton
  InitColdStart provideInitColdStart(AppInfoRepository appInfoRepository) {
    return new InitColdStart(appInfoRepository);
  }
}

File: app/src/main/java/com/kaciula/archiman/injection/DomainModule.kt
package com.kaciula.archiman.injection

import com.kaciula.archiman.domain.boundary.infrastructure.AppRepository
import com.kaciula.archiman.domain.boundary.infrastructure.UserRepository
import com.kaciula.archiman.domain.usecases.GetUsers
import com.kaciula.archiman.domain.usecases.InitColdStart
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class DomainModule {

    @Provides
    @Singleton
    fun provideGetUsers(userRepository: UserRepository): GetUsers {
        return GetUsers(userRepository)
    }

    @Provides
    @Singleton
    fun provideInitColdStart(appRepository: AppRepository): InitColdStart {
        return InitColdStart(appRepository)
    }
}
